alg/teca_damper.cxx

```diff
@@ -0,0 +1,217 @@
+#include "teca_damper.h"
+
+#include "teca_mesh.h"
+#include "teca_array_collection.h"
+#include "teca_variant_array.h"
+#include "teca_metadata.h"
+#include "teca_cartesian_mesh.h"
+
+#include <algorithm>
+#include <iostream>
+#include <deque>
+#include <set>
+#include <cmath>
+#include <complex.h>
+
+using std::deque;
+using std::vector;
+using std::set;
+using std::cerr;
+using std::endl;
+
+//#define TECA_DEBUG
+namespace {
+
+// get the filter ready to be applied in the next steps 
+template <typename coord_t>
+void get_lat_filter(
+    coord_t *filter, const coord_t *lat_input, size_t n_lat_vals,
+    double filter_lat_width)
+{
+    for (size_t i = 0; i < n_lat_vals; ++i) {
+        filter[i] = 1.0 - exp(-1.0 * std::pow(lat_input[i], 2)/(2 * std::pow(filter_lat_width, 2)));
+    }
+}
+
+// damp the input array using inverted gaussian 
+template <typename num_t, typename coord_t>
+void apply_damper(
+    num_t *output, const num_t *input, const coord_t *filter,
+    size_t n_lat_vals, size_t n_lon_vals)
+{
+    for (size_t i = 0; i < n_lat_vals; ++i) {
```

Comment:
this loop and the indexing calculations are correct. however, the convention is to use i for indexing along the x axis and j for indexing the y axis. please do that.